### PR TITLE
(fix) add response headers in fetch client

### LINKED ISSUE #1694: add headers to response
## What are the steps to reproduce this issue?

1. Generate queries by using fetch api
2. Look into the response object
3. No headers are in the response object

## What happens?

No headers will be in the response object. The code looks like:

```
  const res = await fetch(getTaskResourceListAllUrl(params),
  {      
    ...options,
    method: 'GET'
  }

  )
  const data = await res.json()
  // Issue: please add the "res.headers" into the return statement
  return { status: res.status, data }
}
```

## What were you expecting to happen?

Add `headers` in the `response` object, at least when `openapi.json` describes custom headers for an API endpoint.

## What versions are you using?
"orval": "^7.2.0",

### PR DESCRIPTION
**Pull request by melloware**
Fix #1694 add response headers in fetch client

### CODE REVIEW

File: packages/fetch/src/index.ts [hunk 1]
```diff
@@ -167,7 +167,7 @@ ${
   )
   const data = await res.json()
 
-  ${override.fetch.includeHttpStatusReturnType ? 'return { status: res.status, data }' : `return data as ${responseTypeName}`}
+  ${override.fetch.includeHttpStatusReturnType ? 'return { status: res.status, data, headers?: res.headers }' : `return data as ${responseTypeName}`}
```
Comment (soartec-lab) [inline packages/fetch/src/index.ts]: Could you add headers to the response type as well ?

https://github.com/orval-labs/orval/blob/master/packages/fetch/src/index.ts#L112-L116

**Comment by soartec-lab** [comment]
Hi @melloware, Please give me some time to think about this response.

**Comment by melloware** [comment]
Yep converting to draft was just seeing if that is what the OP meant!

File: packages/fetch/src/index.ts (hunk 1 above)
Comment (soartec-lab) [inline packages/fetch/src/index.ts]: We should update the name of `includeHttpStatusReturnType` to something like `includeHttpResponseReturnType`.

**Review by soartec-lab — COMMENTED** [review]
Good update. I have added comments where additional updates are needed.

**Comment by melloware** [comment]
@soartec-lab you are more familiar with this code did you want to submit a PR and I can close this one?

**Comment by melloware** [comment]
OK changes made

**Comment by soartec-lab** [comment]
@melloware 

Thanks!
can you regenerate the  `fetch` client and  update the `custom-fetch` implement in sample apps ?

**Comment by melloware** [comment]
@soartec-lab not sure what you mean by regenerate but i updated `orval.config.js` and ` custom-fetch.js` in samples directory.

**Comment by soartec-lab** [comment]
@melloware 

Sorry for confusing you. To be more precise, I would like you to regenerate the sample app using the following command.

```bash
cd samples
yarn generate-api --force
```

Your change adds `headers` to `returnType`, so the `returnType` of the sample app generated by orval should also be updated.

**Comment by melloware** [comment]
Weird when I run that command in samples I get..

```
 Tasks:    0 successful, 21 total
Cached:    0 cached, 21 total
  Time:    1.957s
Failed:    react-query-basic#generate-api, react-query-form-data#generate-api, react-query-form-url-encoded#generate-api

 ERROR  run failed: command  exited (1)
```

**Comment by melloware** [comment]
Would you like to just merge this PR and then you can take a look?

**Comment by soartec-lab** [comment]
@melloware 
Can I run it and put commits on this branch ?

**Comment by melloware** [comment]
Oh sure thing my branch should be editable for you!

**Comment by soartec-lab** [comment]
@melloware 
Can you try `cherry-pick` this ?

https://github.com/orval-labs/orval/pull/1699/commits/13533d5d1e8ab2776653c642547cfe65419855c3

Or you can switch to this PR.

**Pull request closed without merging**